fix: update API paths to use API domain

CharlieC3 opened the pull request:
This updates any endpoints which use a Stacks Blockchain API path to call the correctly configured endpoint for the service. Previously, this bug was hidden because the Stacks Blockchain API and Stacks Core RPC service had to be configured on the same domain. Now that they can be configured on separate domains since release 1.3.0, this bug popped up.

I've deployed and tested this patch and it appears to be running the same as it were at version 1.2.1.

CharlieC3 committed:
fix: update API paths to use API domain

jcnelson reviewed (changes requested):
Fix the failing tests first

CharlieC3 committed:
ci: fix tests

Set blockstackAPIUrl to coreApiUrl to ensure tests are run against the correct endpoint

CharlieC3 commented:
@jcnelson Fixed!

wileyj commented on tests/src/index.js:
```diff
@@ -9,6 +9,7 @@ import { StacksMainnet } from '@stacks/network'
 nock.disableNetConnect()
 
 bskConfig.network = new StacksMainnet()
+bskConfig.network.blockstackAPIUrl = bskConfig.network.coreApiUrl
```
wouldn't this be easy to change  the env var to `STACKS_API_URL` etc, vs continuing to use the `BLOCKSTACK_API_URL`?

CharlieC3 replied:
Maybe, but it's not a high enough priority for us to invest more resources in at the moment.

wileyj reviewed (approved):
shipit

blockstack-devops commented:
:tada: This PR is included in version 1.3.1 :tada:

The release is available on [GitHub release](https://github.com/stacks-network/subdomain-registrar/releases/tag/v1.3.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
